### PR TITLE
Prepare for `lxc-to-lxd` to not be shipped in the snap

### PR DESCRIPTION
**Pull request by simondeziel**
@tomponline when coping what was done for `lxc` and `lxd-migrate` steps, I initially tried building `lxc-to-lxd` with `CGO_ENABLED=0` but that didn't work. I'm not sure of the implication of not having this.

### CODE REVIEW

**Comment by tomponline** [comment]
> I initially tried building `lxc-to-lxd` with `CGO_ENABLED=0` but that didn't work. I'm not sure of the implication of not having this.

What was the problem? Was it build like this in the snap?

File: doc/howto/migrate_from_lxc.md [hunk 1]
```diff
@@ -16,10 +16,12 @@ However, this tool does not migrate any of the LXC container configuration.
 
 ## Get the tool
 
-If you're using the snap, the `lxc-to-lxd` is automatically installed.
-It is available as `lxd.lxc-to-lxd`.
+To get the `lxc-to-lxd` tool you can either download a pre-built binary:
 
-Otherwise, make sure that you have `go` ({ref}`requirements-go`) installed and get the tool with the following command:
+1. Download the `bin.linux.lxc-to-lxd` tool ([`bin.linux.lxc-to-lxd.aarch64`](https://github.com/canonical/lxd/releases/latest/download/bin.linux.lxc-to-lxd.aarch64) or [`bin.linux.lxc-to-lxd.x86_64`](https://github.com/canonical/lxd/releases/latest/download/bin.linux.lxc-to-lxd.x86_64)) from the **Assets** section of the latest [LXD release](https://github.com/canonical/lxd/releases).
+1. Save the binary as `lxc-to-lxd` and make it executable (usually by running `chmod u+x bin.linux.lxc-to-lxd`).
```
Comment (ru-fu) [inline doc/howto/migrate_from_lxc.md:22]:
```suggestion
1. Save the binary as `lxc-to-lxd` and make it executable (usually by running `chmod u+x lxc-to-lxd`).
```

File: doc/howto/migrate_from_lxc.md [hunk 2]
```diff
@@ -16,10 +16,12 @@ However, this tool does not migrate any of the LXC container configuration.
 
 ## Get the tool
 
-If you're using the snap, the `lxc-to-lxd` is automatically installed.
-It is available as `lxd.lxc-to-lxd`.
+To get the `lxc-to-lxd` tool you can either download a pre-built binary:
```
Comment (ru-fu) [inline doc/howto/migrate_from_lxc.md:19]:
```suggestion
To get the `lxc-to-lxd` tool, you can download a pre-built binary:
```

Not a fan of either/or with so much text in between. ;) But it should work fine without.

File: doc/howto/migrate_from_lxc.md [hunk 3]
```diff
@@ -16,10 +16,12 @@ However, this tool does not migrate any of the LXC container configuration.
...
 
-Otherwise, make sure that you have `go` ({ref}`requirements-go`) installed and get the tool with the following command:
+1. Download the `bin.linux.lxc-to-lxd` tool ([`bin.linux.lxc-to-lxd.aarch64`](https://github.com/canonical/lxd/releases/latest/download/bin.linux.lxc-to-lxd.aarch64) or [`bin.linux.lxc-to-lxd.x86_64`](https://github.com/canonical/lxd/releases/latest/download/bin.linux.lxc-to-lxd.x86_64)) from the **Assets** section of the latest [LXD release](https://github.com/canonical/lxd/releases).
+1. Save the binary as `lxc-to-lxd` and make it executable (usually by running `chmod u+x bin.linux.lxc-to-lxd`).
+
+Or if you have `go` ({ref}`requirements-go`) installed you can get the tool with the following command:
```
Comment (ru-fu) [inline doc/howto/migrate_from_lxc.md:24]:
```suggestion
If you have `go` ({ref}`requirements-go`) installed, you can build the tool with the following command:
```

**Comment by simondeziel** [comment]
It seems I'll need to have the `lxc-to-lxd` binary statically linked to avoid depending on libdqlite.so and liblxc.so being available on the target system.

This will also be more challenging to provide a binary for arm64.

**Comment by tomponline** [comment]
Ah in that case lets leave in the snap for now.

**Comment by tomponline** [comment]
@simondeziel it would be good if we could refactor lxc-to-lxd to use the `lxc-*` tooling of the system if possible, then we wouldn't need to link to liblxc.

**Comment by simondeziel** [comment]
Closing in favor of #12990

**Pull request closed without merging**